lldb/source/Plugins/Process/AIX/NativeProcessAIX.cpp

```diff
@@ -0,0 +1,265 @@
+  if (m_terminal_fd != -1) {
+    Status status = EnsureFDFlags(m_terminal_fd, O_NONBLOCK);
+    assert(status.Success());
+  }
+
+  // Let our process instance know the thread has stopped.
+  SetCurrentThreadID(tids[0]);
+  SetState(StateType::eStateStopped, false);
+}
+
+llvm::Expected<std::vector<::pid_t>> NativeProcessAIX::Attach(::pid_t pid) {
+
+  Error status;
+  if ((status = PtraceWrapper(PT_ATTACH, pid)).Fail()) {
+    return errorCodeToError(errnoAsErrorCode());
+  }
+
+  int wpid = llvm::sys::RetryAfterSignal(-1, ::waitpid, pid, nullptr, WNOHANG);
+  if (wpid <= 0) {
+    return llvm::errorCodeToError(
+        std::error_code(errno, std::generic_category()));
+  }
+  LLDB_LOG(log, "adding pid = {0}", pid);
+
+  std::vector<::pid_t> tids;
+  tids.push_back(pid);
+  return std::move(tids);
+}
+
+void NativeProcessAIX::MonitorSIGTRAP(const WaitStatus status,
+                                      NativeThreadAIX &thread) {}
+
+void NativeProcessAIX::MonitorBreakpoint(NativeThreadAIX &thread) {}
+
+bool NativeProcessAIX::SupportHardwareSingleStepping() const { return false; }
+
+Status NativeProcessAIX::Resume(const ResumeActionList &resume_actions) {
+  return Status();
+}
+
+Error NativeProcessAIX::Halt() {
+  Error error;
+  return error;
+}
+
+Error NativeProcessAIX::Detach() {
+  Error error;
+  return error;
+}
+
+Error NativeProcessAIX::Signal(int signo) {
+  Error error;
+  return error;
+}
+
+Error NativeProcessAIX::Interrupt() { return Status(); }
+
+Error NativeProcessAIX::Kill() {
+  Error error;
+  return error;
+}
+
+Error NativeProcessAIX::SetBreakpoint(lldb::addr_t addr, uint32_t size,
+                                      bool hardware) {
+  if (hardware)
+    return SetHardwareBreakpoint(addr, size);
+  else
+    return SetSoftwareBreakpoint(addr, size);
+}
+
+Error NativeProcessAIX::RemoveBreakpoint(lldb::addr_t addr, bool hardware) {
+  if (hardware)
+    return RemoveHardwareBreakpoint(addr);
+  else
+    return NativeProcessProtocol::RemoveBreakpoint(addr);
+}
+
+int8_t NativeProcessAIX::GetSignalInfo(WaitStatus wstatus) const {
+  return wstatus.status;
+}
+
+Error NativeProcessAIX::Detach(lldb::tid_t tid) {
+  if (tid == LLDB_INVALID_THREAD_ID)
+    return Status();
+
+  return PtraceWrapper(PT_DETACH, tid);
+}
+
+// Wrapper for ptrace to catch errors and log calls. Note that ptrace sets
+// errno on error because -1 can be a valid result (i.e. for PTRACE_PEEK*)
+Error NativeProcessAIX::PtraceWrapper(int req, lldb::pid_t pid, void *addr,
+                                      void *data, size_t data_size,
+                                      long *result) {
+  Error error;
+  long int ret;
+
+  Log *log = GetLog(POSIXLog::Ptrace);
+  errno = 0;
+  if (req < PT_COMMAND_MAX) {
+    if (req == PT_ATTACH) {
+      ptrace64(req, pid, 0, 0, nullptr);
+    } else if (req == PT_DETACH) {
+      ptrace64(req, pid, 0, 0, nullptr);
+    }
+  } else {
+    assert(0 && "Not supported yet.");
+  }
```

Comment:
On the subject of removing braces, I would write this as:
```
  if (req < PT_COMMAND_MAX) {
    if (req == PT_ATTACH)
      ptrace64(req, pid, 0, 0, nullptr);
    else if (req == PT_DETACH)
      ptrace64(req, pid, 0, 0, nullptr);
  } else
    assert(0 && "Not supported yet.");
```
But, I think you are missing an else here. Perhaps you are going to expand this to recognise more PT_ constants later, and that's why it looks incomplete right now?

As far as I can tell right now it's:
```
switch (req) {
  case PT_ATTACH:
  case PT_DETATCH:
    ptrace64(req, pid, 0, 0, nullptr);
    break;
  default:
    assert(0 && "PT_ request not supported yet.");
}
```
But if that's not going to scale as you add more PT_ values, I suppose we can take this as is knowing that it will make sense later.